toolchain/check/impl.cpp

```diff
@@ -84,6 +84,10 @@ auto ImplWitnessForDeclaration(Context& context, const SemIR::Impl& impl)
     -> SemIR::InstId {
   CARBON_CHECK(!impl.has_definition_started());
 
+  if (impl.self_id == SemIR::ErrorInst::SingletonInstId) {
```

Comment:
Noticed this was wrong. It's a NameRef not an error, it's the type that becomes an error. Fixed in last commit.